src/csharp/Grpc.Core/GrpcEnvironment.cs

```diff
@@ -311,13 +328,16 @@ internal static void GrpcNativeShutdown()
         /// </summary>
         private async Task ShutdownAsync()
         {
-            if (isClosed)
+            if (isShutdown)
```

Comment:
this is technically not thread safe either, but I'm not sure how important it really is if multiple threads dont call shutdown.
A switch to interlocked.exchange and inspect the result would fix.